tests/foreman/api/test_repository.py

```diff
@@ -103,6 +115,80 @@ def test_positive_create_with_name(self):
                     product=self.product, name=name).create()
                 self.assertEqual(name, repo.name)
 
+    @tier1
+    def test_positive_create_with_http_proxy(self):
+        """Create a repository with http_proxy.
+
+        :id: 5b3b992e-02d3-4b16-95ed-21f1588c7741
+
+        :expectedresults: A repository is created with the given http_proxy.
+
+        :CaseImportance: Critical
+        """
+        # With selected_http_proxy
+        repo = entities.Repository(
+            product=self.product,
+            http_proxy_policy='use_selected_http_proxy',
+            http_proxy_id=self.http_proxy.id
+        ).create()
+        assert repo.http_proxy_policy == 'use_selected_http_proxy'
+        assert repo.http_proxy_id == self.http_proxy.id
+        # With global_default_http_proxy
+        repo = entities.Repository(
+            product=self.product,
+            http_proxy_policy='global_default_http_proxy',
+        ).create()
+        assert repo.http_proxy_policy == 'global_default_http_proxy'
+        # With no http_proxy
+        repo = entities.Repository(
+            product=self.product,
+            http_proxy_policy='none',
+        ).create()
+        assert repo.http_proxy_policy == 'none'
+        # Update http_proxy for repo
+        updated_repo = entities.Repository(
+            id=repo.id,
+            http_proxy_policy='use_selected_http_proxy',
+            http_proxy_id=self.http_proxy.id
+        ).create()
+        assert updated_repo.http_proxy_policy == 'use_selected_http_proxy'
+        assert updated_repo.http_proxy_id == self.http_proxy.id
+
+    @tier2
+    def test_positive_assign_http_proxy_to_products(self):
```

Comment:
Is this test not a part of `test_product.py`? also can this to be a `CRUD` as well ?